lib/configproxy.js

```diff
@@ -222,20 +232,25 @@ ConfigurableProxy.prototype.get_routes = function (req, res) {
         }
     }
     res.writeHead(200, { 'Content-Type': 'application/json' });
-    var routes = {};
-    if (inactive_since) {
-        Object.keys(this.routes).map(function (path) {
-            var route = that.routes[path];
-            if (route.last_activity < inactive_since) {
-                routes[path] = route;
-            }
-        });
-    } else {
-        routes = this.routes;
-    }
-    res.write(JSON.stringify(routes));
-    res.end();
-    this.statsd.increment('api.route.get', 1);
+
+    this._routes.getAll(function (routes) {
+        var results = {};
+
+        if (inactive_since) {
+            var keys = Object.keys(routes).filter(function (key) {
+                return routes[key].last_activity < inactive_since;
```

Comment:
The previous implementation was more efficient because it iterated over the routes only once, rather than twice.